### ISSUE TITLE
Go to definition not working in network drive after updating Pylance from 2023.11.10 to 2023.12.1

### ISSUE THREAD

**Issue by BradXiao** [microsoft/pylance-release#5260]
## Environment data

-   Language Server version: Pylance language server 2023.11.10  (works fine)
-   Language Server version: Pylance language server 2023.12.1  (doesn't work)
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.12(Anaconda virtual environment)
-  A network drive T:\\ pointing to \\\\192.168.22.2.

## Code Snippet

```python
# The files are in a network driver T:\\ (\\\\192.168.22.2).
# file pkg.py
def test_function():
    pass

# file test.py
from .pkg import test_function
test_function()
```

## Repro Steps

1. Open test.py in code snippet
2. Go to definition of test_function()

## Expected behavior

Navigate to test_function() in the file pkg.py.

## Actual behavior

Error message.

```
The editor could not be opened due to an unexpected error: Unable to read file '\\192.168.22.2\test\pkg.py' (Unknown (FileSystemError): UNC host '192.168.22.2' access is not allowed. Please update the 'security.allowedUNCHosts' setting if you want to allow this host.)
```

## Logs

```
2023-12-14 13:04:51.945 [info] (28384) [BG(1)] getSemanticTokens delta previousResultId:1702530282604 at \\192.168.22.2\test\test.py (0ms)
2023-12-14 13:05:01.814 [info] (28384) [BG(1)] getSemanticTokens delta previousResultId:1702530282604 at \\192.168.22.2\test\test.py (0ms)
```

**Comment by heejaechang** [comment]
@BradXiao I can't repro. it works as expected. does this only happen if network drive is accessed through ip?

**Comment by BradXiao** [comment]
It is an opened directory with the path T:\test and the drive T's real path is \\\\192.168.22.2\\brad. I did not open the path with IP directly. All the file paths shown in vscode are like T:\\test\\???.py. But when it comes to go to definition, it resolved as \\\\192.168.22.2\\brad\\test\\???.py.
It only happened in pylance version 2023.12.1.  Using pylance 2023.11.10 with the exact the same environment and directory worked well.

The attached file is the log file. Hope it helps. Thanks.

[logs.txt](https://github.com/microsoft/pylance-release/files/13695127/logs.txt)

**Comment by heejaechang** [comment]
thank you, I can repro once I did map the drive!

**Comment by heejaechang** [comment]
it is dup of https://github.com/microsoft/pyrx/issues/4272

**Comment by heejaechang** [comment]
we are working on a fix but it will probably release after holiday season.

**Comment by rchiodo** [comment]
This should be fixed when our next prerelease goes out (usually on Wednesdays).

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.12.104, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202312104-22-January-2024-PreRelease)
